src/Symfony/Component/BrowserKit/AbstractBrowser.php

```diff
@@ -333,6 +333,11 @@ public function submitForm(string $button, array $fieldValues = [], string $meth
         }
 
         $buttonNode = $this->crawler->selectButton($button);
+
+        if (0 === $buttonNode->count()) {
+            throw new \InvalidArgumentException(sprintf('The form doesn\'t contain any button with "%s" as its content, id, value or name.', $button));
```

Comment:
@alamirault thanks a lot for working on this!

After proposing the new error message in #49826 I realized that I was missing that a page can contain more than one form. So, I think we should tweak the message like:

```
There are no forms that contain a button with ...
```

Let's hear if others have other suggestions to make this message as clear as possible. Thanks!

Reply:
Note that this query looks for buttons even outside forms. If the button with than name is not in a form, the failure will be in the `->form()` method.

This should probably say `There is no button with...` here

Reply:
Agreed with you @stof. I updated PR